interfaces/Portal/src/app/services/programs-service-api.service.ts

```diff
@@ -36,26 +36,29 @@ import { arrayToXlsx } from '../shared/array-to-xlsx';
 import { ApiService } from './api.service';
 import { PaginationFilter, PaginationSort } from './filter.service';
 
+import { UsersService } from '121-service-angular-api';
+
 @Injectable({
   providedIn: 'root',
 })
 export class ProgramsServiceApiService {
-  constructor(private apiService: ApiService) {}
+  constructor(
+    private apiService: ApiService,
+    private usersService: UsersService,
+  ) {}
 
   login(username: string, password: string): Promise<User | null> {
     console.log('ProgramsService : login()');
 
-    return this.apiService
-      .post(
-        environment.url_121_service_api,
-        ApiPath.usersLogin,
-        {
-          username,
-          password,
-        },
-        true,
-      )
+    return this.usersService
+      .userControllerLogin({
+        username,
+        password,
+      })
+      .toPromise()
       .then((response) => {
+        // this works
+        // but response is typed as 'any' 🙈
```

Comment:
The biggest shortcoming of this approach - it doesn't actually infer types properly. You can see that this is explicitly mis-typed in [`users.service.ts`](https://github.com/global-121/121-platform/pull/5490/files#diff-684af501228b7f3a61c11b131a41436e13c7e73ab9ccde2f16f07ee6390ce28fR517-R520), where the return type is typed as `any`.

So it kind of defeats the purpose of doing this whole thing.